File: token-validation-core/src/main/java/no/nav/security/token/support/core/jwt/JwtToken.java
package no.nav.security.token.support.core.jwt;

import com.nimbusds.jwt.JWT;
import com.nimbusds.jwt.JWTClaimsSet;
import com.nimbusds.jwt.JWTParser;

import java.text.ParseException;

public class JwtToken {

    private final String encodedToken;
    private final JWT jwt;
    private final JwtTokenClaims jwtTokenClaims;

    public JwtToken(String encodedToken) {
        try {
            this.encodedToken = encodedToken;
            this.jwt = JWTParser.parse(encodedToken);
            this.jwtTokenClaims = new JwtTokenClaims(getJwtClaimsSet());
        } catch (ParseException e) {
            throw new RuntimeException(e);
        }
    }

    public String getIssuer() {
        return jwtTokenClaims.getIssuer();
    }

    public String getSubject() {
        return jwtTokenClaims.getSubject();
    }

    public String getTokenAsString() {
        return encodedToken;
    }

    public boolean containsClaim(String name, String value) {
        return jwtTokenClaims.containsClaim(name, value);
    }

    public JwtTokenClaims getJwtTokenClaims() {
        return jwtTokenClaims;
    }

    protected JWT getJwt() {
        return jwt;
    }

    private JWTClaimsSet getJwtClaimsSet() throws ParseException {
        return jwt.getJWTClaimsSet();
    }
}

File: token-validation-core/src/main/kotlin/no/nav/security/token/support/core/jwt/JwtToken.kt
package no.nav.security.token.support.core.jwt

import com.nimbusds.jwt.JWT
import com.nimbusds.jwt.JWTParser
import com.nimbusds.jwt.SignedJWT
import kotlin.DeprecationLevel.WARNING

open class JwtToken(val encodedToken : String, protected val jwt : JWT, val jwtTokenClaims : JwtTokenClaims) {
    constructor(encodedToken : String) : this(encodedToken, JWTParser.parse(encodedToken), JwtTokenClaims(JWTParser.parse(encodedToken).jwtClaimsSet))

    val jwtClaimsSet = jwt.jwtClaimsSet

    val subject  = jwtTokenClaims.subject

    val issuer = jwtTokenClaims.issuer

    @Deprecated("Use getEncodedToken instead", ReplaceWith("getEncodedToken()"), WARNING)
    val tokenAsString = encodedToken

    fun asBearer() = "Bearer $encodedToken"

    fun containsClaim(name : String, value : String)  = jwtTokenClaims.containsClaim(name, value)

    companion object {
         fun SignedJWT.asBearer() = "Bearer ${serialize()}"
    }
}